### PR TITLE
Add queso-config

### PR DESCRIPTION
**Pull request by pbauman**
Very much akin to the libmesh-config shell script in libMesh. This will provide necessary information for building and linking a QUESO application.

### CODE REVIEW

File: src/apps/queso-config.in [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"--include")
+	    return_val="-I${includedir} $return_val"
+	    ;;
+
+	"--libs")
+	    return_val="-lqueso $return_val"
+	    ;;
+
+	"--ldflags")
+	    return_val="-Wl,-rpath,${libdir} -L${libdir} $return_val"
```
Comment (dmcdougall) [inline src/apps/queso-config.in]: Yes, I like the `rpath`.

**Comment by dmcdougall** [comment]
@pbauman Is this the output you expect?

```
$ ~/queso/bin/queso-config --cxx
mpic++
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --cc
mpicc
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --fc
gfortran
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --fflags
@FFLAGS@
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --version
0.54.0
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --host
x86_64-apple-darwin14.5.0
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --ldflags
-Wl,-rpath,/Users/damon/queso/lib -L/Users/damon/queso/lib
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --libs
-lqueso
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --include
-I/Users/damon/queso/include
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --cxxflags
-g -O2 -Wall
$damon@quagmire:((ca1a190...)) ~/git/github/queso
$ ~/queso/bin/queso-config --cppflags

```

Note, `--fflags` and `--cppflags`.  Should `--cppflags` basically return the same thing as `--include`?

**Comment by pbauman** [comment]
Rebased. @dmcdougall re: cppflags --- I'm adhering to what's done in libMesh and having `--include` be the `-I` statements and the `--cppflags` be the `-D` statements. I hadn't tested it (e.g. turning on --enable-debug), so I went back and did and, of course, it didn't work. I popped a commit on that fixes that.

re: `--fcflags`, that output is because `FCFLAGS` never gets set. I leave it to you as to what to do. I'd say drop just remove the fc bits completely from the `queso-config`.

**Comment by dmcdougall** [comment]
> I leave it to you as to what to do. I'd say drop just remove the fc bits completely from the `queso-config`.

Fine by me.

**Comment by pbauman** [comment]
Fortran bits removed.

**Comment by dmcdougall** [comment]
Thanks.

**Pull request merged**